### ISSUE TITLE
Crashes

### ISSUE THREAD

**Issue by DThielke** [Kapps/PoEUncrasher#8]
I'm getting the following error when trying to run this. I have confirmed I have .NET9 installed, I'm NOT running from the zip, and I've tried running as admin and as a local user.

```
09:15:45 info: Program[0] Waiting for Path of Exile process to launch before doing anything.

Unhandled exception. System.ComponentModel.Win32Exception (87): The parameter is incorrect.
   at System.Diagnostics.Process.set_ProcessorAffinityCore(IntPtr value)
   at System.Diagnostics.Process.set_ProcessorAffinity(IntPtr value)
   at Program.<>c__DisplayClass0_0.<<<Main>$>g__ParkCores|6>d.MoveNext() in C:\Users\opant\RiderProjects\PoEUncrasher\PoEUncrasher\Program.cs:line 158
--- End of stack trace from previous location ---
   at Program.<Main>$(String[] args) in C:\Users\opant\RiderProjects\PoEUncrasher\PoEUncrasher\Program.cs:line 101
   at Program.<Main>(String[] args)
```

**Comment by doodaz** [comment]
Same is happening to me on both versions of the program. I'm on a 5950x with Windows 10 22H2. Tried parking 8 or 4 cores.

**Comment by Kapps** [comment]
Hmm, my suspicion is this is related to game mode on the _950X CPUs. When setting the affinity, the program has to pass a bit-field for every core indicating whether it's enabled or not. So for 8 cores / 16 threads, something like `1111111111111100` to enable the last 14 threads and disable the first 2. It uses the `NUMBER_OF_PROCESSORS` environment variable to determine the number of threads available. If the wrong number of threads is passed in, this error would occur. I bet that game mode causes this to report a different number.

If you open up `cmd` and type `echo %NUMBER_OF_PROCESSORS%`, what does it say?

**Comment by doodaz** [comment]
Hi Kapps, thanks for the quick reply. As I said I'm on a 5950x so that's 16 cores / 32 threads:

C:\>echo %NUMBER_OF_PROCESSORS%
32

**Comment by DThielke** [comment]
I have a 7950x3d - also 16 cores / 32 threads.

**Comment by Kapps** [comment]
I added a new pre-release with a theoretical workaround (also hopefully a fix for the virus scanner false positives): https://github.com/Kapps/PoEUncrasher/releases/tag/v1.2.1

I can't replicate this on my CPU as I think it's specifically the _950X chips, and particularly with Game Mode. The tool will now try to use its own process affinity to calculate the amount of cores rather than relying on what Windows reports to us. This might not work because if it is a game mode related issue, this tool wouldn't be considered a game and therefore maybe it has different settings than PoE would.

Basically, let me know if it works!

**Comment by nura196** [comment]
Hi,
It seems to have not resolved for me. Getting below error. Number of processors: `32` , `Intel(R) Core(TM) i9-14900HX   2.20 GHz`

```
09:54:34 warn: Program[0] Number of cores reported to us was 32, but this appears different than the affinity. Trying to calculate real number.
Unhandled exception. System.OverflowException: Value was either too large or too small for a UInt32.
   at System.Number.ThrowOverflowException[TInteger]()
   at System.ParseNumbers.GrabInts(Int32 radix, ReadOnlySpan`1 s, Int32& i, Boolean isUnsigned)
   at System.ParseNumbers.StringToInt(ReadOnlySpan`1 s, Int32 radix, Int32 flags, Int32& currPos)
   at System.Convert.ToInt32(String value, Int32 fromBase)
   at Program.<>c__DisplayClass0_0.<<Main>$>g__GetNumberProcessors|9() in C:\Users\opant\RiderProjects\PoEUncrasher\PoEUncrasher\Program.cs:line 226
   at Program.<Main>$(String[] args) in C:\Users\opant\RiderProjects\PoEUncrasher\PoEUncrasher\Program.cs:line 11
   at Program.<Main>(String[] args)
   ```

**Comment by Kapps** [comment]
Updated the zip file to fix that issue.

**Comment by nura196** [comment]
> Updated the zip file to fix that issue.

Thanks for quick fix!
It seems to have resolved it. Will let it you know if it doesn't crash for me

```
10:18:05 info: Program[0] Waiting for Path of Exile process to launch before doing anything.
10:18:22 info: Program[0] Parked cores: 11111111111111111111111111111100
10:18:22 info: Program[0] Reading client data from C:\Program Files (x86)\Steam\steamapps\common\Path of Exile 2\logs\client.txt
10:18:49 info: Program[0] Parked cores: 11111111111111111111111111111100
10:19:06 info: Program[0] Unparked cores: 11111111111111111111111111111111
```

**Comment by nura196** [comment]
Original issue is resolved and program is running fine. But it still crash and PC lock for me. Maybe it solves for others.
Looks like it is not getting triggered for all page loads. Attached client.txt for that [timeframe.]([url]([client.txt](https://github.com/user-attachments/files/18220845/client.txt)))

* Logs
```
10:41:52 info: Program[0] Waiting for Path of Exile process to launch before doing anything.
10:41:52 info: Program[0] Reading client data from C:\Program Files (x86)\Steam\steamapps\common\Path of Exile 2\logs\client.txt
10:42:27 info: Program[0] Parked cores: 11111111111111111111111111111100
10:42:41 info: Program[0] Unparked cores: 11111111111111111111111111111111
10:43:29 info: Program[0] Parked cores: 11111111111111111111111111111100
10:43:39 info: Program[0] Unparked cores: 11111111111111111111111111111111
```

**Comment by nura196** [comment]
Attaching client.txt
[client.txt](https://github.com/user-attachments/files/18220847/client.txt)
